src/Runner/Game.java

```diff
@@ -3,99 +3,71 @@
-		HashMap<String, BasePlayer> createsCharacter = new HashMap<>();
-		Names[] names = {Names.MAGE,Names.MARKSMAN,Names.SWORDSMAN,Names.TANK};
-		
-		if(!player) {
-			String enemyName = "Enemy";
-			String enemyWeapon = "Sparkle Glitter";
-			
-			Random number = new Random();
-			int value = number.nextInt(Names.values().length);
-			String key = Names.values()[value].toString();
-			BasePlayer enemy = createsCharacter.get(key);
-			enemy.setName(enemyName);
-			enemy.setWeaponName(enemyWeapon);
-			
-			return enemy;
+	public BasePlayer createPlayer() {
+		final String profession = JOptionPane.showInputDialog(null,
+					"Pick a class!", "Class Selection",
+					JOptionPane.QUESTION_MESSAGE, null,
+					Professions.ALLOWED_PROFESSION_NAMES.toArray(), "select a class").toString();
+		if (!Professions.isValidProfession(profession)) {
+			throw new IllegalStateException("Invalid profession entered. This should never happen.");
 		}
 		
+		final String playerName = JOptionPane.showInputDialog(null, "Enter the name of your " + profession);
+		final String weaponName = JOptionPane.showInputDialog(null, "Enter the name of your Weapon");
 		
-		
-		
-		String classSelection = (String) JOptionPane.showInputDialog(null,"Pick a class!", "Class Selection",JOptionPane.QUESTION_MESSAGE,null,names,"select a class").toString();
-		String name = (String) JOptionPane.showInputDialog(null,"Enter the name of your " + classSelection);
-		String nameOfWeapon = (String) JOptionPane.showInputDialog(null,"Enter the name of your Weapon");
-		
-		BiFunction<String,String,Swordsman> createSwordsman = Swordsman::new;
-		BiFunction<String,String,Marksman> createMarksman = Marksman::new;
-		BiFunction<String,String,Tank> createTank = Tank::new;
-		BiFunction<String,String,Mage> createMage = Mage::new;
-			
-		createsCharacter.put(Names.SWORDSMAN.toString(), (BasePlayer) createSwordsman.apply(name,nameOfWeapon));
-		createsCharacter.put(Names.MARKSMAN.toString(), (BasePlayer) createMarksman.apply(name, nameOfWeapon));
-		createsCharacter.put(Names.TANK.toString(), (BasePlayer) createTank.apply(name, nameOfWeapon));
-		createsCharacter.put(Names.MAGE.toString(), (BasePlayer) createMage.apply(name, nameOfWeapon));
-			
-		BasePlayer character = createsCharacter.get(classSelection);
-
-		return character;
-		
+		return Professions.getPlayerForProfession(profession, playerName, weaponName);
 	}
 	
-
-	public static void fight(BasePlayer player, BasePlayer Enemy) {
-
-		System.out.print(player.getName() + " vs " + Enemy.getName());
+	public BasePlayer createEnemy() {
+		return Professions.getPlayerForProfession(Professions.getRandomProfession(this.rng), DEFAULT_ENEMY_NAME, DEFAULT_ENEMY_WEAPON_NAME);
+	}
+	
+	public void fight(final BasePlayer player, final BasePlayer enemy) {
+		System.out.print(player.getName() + " vs " + enemy.getName());
 		
-		int round = 1;
-		while(round > 0){
-			while(player.getHealth() > 0 || Enemy.getHealth() > 0){

```

Comment:
This should've been an && statement. Because if player's health is `-400` and enemy's health is `1000`, the game would keep going until enemy's health was negative as well. Wouldn't have made sense.